itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java

```diff
@@ -331,15 +350,56 @@ private void printOutput() throws IOException {
   }
 
   public Database getDatabase(String dbName) throws Exception {
-    return client.getDatabase(dbName);
+    try {
+      return client.getDatabase(dbName);
+    } catch (NoSuchObjectException e) {
+      return null;
+    }
+  }
+
+  public List<String> getAllTables(String dbName) throws Exception {
+    return client.getAllTables(dbName);
   }
 
   public Table getTable(String dbName, String tableName) throws Exception {
-    return client.getTable(dbName, tableName);
+    try {
+      return client.getTable(dbName, tableName);
+    } catch (NoSuchObjectException e) {
+      return null;
+    }
+  }
+
+  public List<Partition> getAllPartitions(String dbName, String tableName) throws Exception {
+    try {
+      return client.listPartitions(dbName, tableName, Short.MAX_VALUE);
+    } catch (NoSuchObjectException e) {
+      return null;
+    }
   }
 
   public Partition getPartition(String dbName, String tableName, List<String> partValues) throws Exception {
-    return client.getPartition(dbName, tableName, partValues);
+    try {
+      return client.getPartition(dbName, tableName, partValues);
+    } catch (NoSuchObjectException e) {
+      return null;
+    }
+  }
+
+  public List<SQLPrimaryKey> getPrimaryKeyList(String dbName, String tblName) throws Exception {
+    return client.getPrimaryKeys(new PrimaryKeysRequest(dbName, tblName));
```

Comment:
why catching NoSuchObjectException for getPartition and not for getPrimaryKeyList ?

Reply:
getPrimaryKeyList won't throw it. Instead return empty list.